File: packages/language-ruby/tests/fixtures/sample.rb
module Authentication
  class TokenService
    def initialize(secret_key)
      @secret_key = secret_key
      @revoked_tokens = {}
    end

    def verify_token(token)
      raise ArgumentError, "Token cannot be empty" if token.nil? || token.empty?
      decode_token(token)
    end

    def revoke_token(token)
      @revoked_tokens[token] = Time.now
    end

    def token_revoked?(token)
      @revoked_tokens.key?(token)
    end

    def self.from_env
      new(ENV.fetch("SECRET_KEY"))
    end

    private

    def decode_token(token)
      { sub: "user123", exp: 9_999_999_999 }
    end
  end
end

File: packages/language-php/tests/fixtures/sample.php
<?php

namespace App\Services;

interface AuthInterface
{
    public function verify(string $token): array;
}

class TokenService implements AuthInterface
{
    private string $secretKey;
    private array $revokedTokens = [];

    public function __construct(string $secretKey)
    {
        $this->secretKey = $secretKey;
    }

    public function verify(string $token): array
    {
        if (empty($token)) {
            throw new \InvalidArgumentException('Token cannot be empty');
        }
        return $this->decode($token);
    }

    public function revoke(string $token): void
    {
        $this->revokedTokens[$token] = time();
    }

    private function decode(string $token): array
    {
        return ['sub' => 'user123', 'exp' => 9999999999];
    }
}

function create_token(string $subject): string
{
    return base64_encode($subject . ':' . time());
}
